cluster/src/dunit/scala/org/apache/spark/sql/TPCHDUnitTest.scala

```diff
@@ -77,6 +78,259 @@ class TPCHDUnitTest(s: String) extends ClusterManagerTestBase(s)
+          val parameters = TPCH_Queries.getQ17Parameter(isDynamic)
+          prepStatement.setString(1, "Brand#" + parameters(0) + parameters(1))
+          prepStatement.setString(2, parameters(2))
+        }
+        case 18 => {
+          println("Executing query#18")
+          prepStatement = conn.prepareStatement(TPCH_Queries.getQuery18ForPrepStatement)
+          val parameters = TPCH_Queries.getQ18Parameter(isDynamic)
+          prepStatement.setString(1, parameters(0))
+        }
+        case 19 => {
+          println("Executing query#19")
+          prepStatement = conn.prepareStatement(TPCH_Queries.getQuery19ForPrepStatement)
+          val parameters = TPCH_Queries.getQ19Parameter(isDynamic)
+          prepStatement.setString(1, "Brand#" + parameters(0))
+          prepStatement.setString(2, parameters(1))
+          prepStatement.setString(3, parameters(2))
+          prepStatement.setString(4, "Brand#" + parameters(3))
+          prepStatement.setString(5, parameters(4))
+          prepStatement.setString(6, parameters(5))
+          prepStatement.setString(7, "Brand#" + parameters(6))
+          prepStatement.setString(8, parameters(7))
+          prepStatement.setString(9, parameters(8))
+        }
+        case 20 => {
+          println("Executing query#20")
+          prepStatement = conn.prepareStatement(TPCH_Queries.getQuery20ForPrepStatement)
+          val parameters = TPCH_Queries.getQ20Parameter(isDynamic)
+          prepStatement.setString(1, parameters(0) + "%")
+          prepStatement.setString(2, parameters(1))
+          prepStatement.setString(3, parameters(2))
+          prepStatement.setString(4, parameters(3))
+        }
+        case 21 => {
+          println("Executing query#21")
+          prepStatement = conn.prepareStatement(TPCH_Queries.getQuery21ForPrepStatement)
+          var parameters = TPCH_Queries.getQ21Parameter(isDynamic)
+          prepStatement.setString(1, parameters(0))
+        }
+        case 22 => {
+          println("Executing query#22")
+          prepStatement = conn.prepareStatement(TPCH_Queries.getQuery22ForPrepStatement)
+          var parameters = TPCH_Queries.getQ22Parameter(isDynamic)
+          prepStatement.setString(1, parameters(0))
+          prepStatement.setString(2, parameters(1))
+          prepStatement.setString(3, parameters(2))
+          prepStatement.setString(4, parameters(3))
+          prepStatement.setString(5, parameters(4))
+          prepStatement.setString(6, parameters(5))
+          prepStatement.setString(7, parameters(6))
+          prepStatement.setString(8, parameters(7))
+          prepStatement.setString(9, parameters(8))
+          prepStatement.setString(10, parameters(9))
+          prepStatement.setString(11, parameters(10))
+          prepStatement.setString(12, parameters(11))
+          prepStatement.setString(13, parameters(12))
+          prepStatement.setString(14, parameters(13))
```

Comment:
Generic comment - May be for few queries instead of using setString, we can use setObject or other setter methods as per the data types and then execute.
Please make sure, as discussed, that other kinds of setter combinations are covered elsewhere in the tests. Please update this PR with some information on those.

Reply:
Done, now using other setter methods in some of the queries. Other tests in PreparedQueryRoutingSingleNodeSuite and PreparedQueryRoutingDUnitTest cover  setter methods such as setInt, setDouble, setString, setTimestamp, setDate, setBigDecimal, setLong.